src/GogeToken.sol

```diff
@@ -596,346 +584,350 @@ contract DogeGaySon is ERC20, Ownable {
 
-        uniswapV2Router = _uniswapV2Router;
         uniswapV2Pair = _uniswapV2Pair;
 
         _setAutomatedMarketMakerPair(_uniswapV2Pair, true);
         
+        // exclude from paying dividends to the team wallets and dead addresses
         cakeDividendTracker.excludeFromDividends(address(cakeDividendTracker));
-        
         cakeDividendTracker.excludeFromDividends(address(this));
-        cakeDividendTracker.excludeFromDividends(address(_uniswapV2Router));
-        cakeDividendTracker.excludeFromDividends(deadAddress);
+        cakeDividendTracker.excludeFromDividends(address(uniswapV2Router));
+        cakeDividendTracker.excludeFromDividends(DEAD_ADDRESS);
+        cakeDividendTracker.excludeFromDividends(address(0));
         cakeDividendTracker.excludeFromDividends(owner());
+        cakeDividendTracker.excludeFromDividends(devWallet);
+        cakeDividendTracker.excludeFromDividends(marketingWallet);
+        cakeDividendTracker.excludeFromDividends(teamWallet);
 
         // exclude from paying fees or having max transaction amount
         isExcludedFromFees[marketingWallet] = true;
         isExcludedFromFees[teamWallet] = true;
+        isExcludedFromFees[devWallet] = true;
         isExcludedFromFees[address(this)] = true;
         isExcludedFromFees[owner()] = true;
+        isExcludedFromFees[DEAD_ADDRESS] = true;
+        isExcludedFromFees[address(0)] = true;
         
-        /*
-            _mint is an internal function in ERC20.sol that is only called here
-            and for migration, and it CANNOT be called for any other reason. The 
-            external capability does NOT exist.
-        */
         _mint(owner(), _totalSupply * (10**18));
-        //_mint(owner(), 100000000000 * (10**18));
     }
 
     receive() external payable {
 
     }
 
-    function setDAO(address _dao) external {
-        require(_msgSender() == DAO || _msgSender() == owner(), "Not authorized");
-        require(DAO != _dao, "This address is already the DAO");
-        DAO = _dao;
+    function setGogeDao(address _gogeDao) external {
+        require(_msgSender() == gogeDao || _msgSender() == owner(), "GogeToken.sol::setgogeDao() msg.sender is not owner or gogeDao");
+        require(gogeDao != _gogeDao, "GogeToken.sol::setgogeDao() address is already set");
+
+        gogeDao = _gogeDao;
+
+        isExcludedFromFees[gogeDao] = true;
+        cakeDividendTracker.excludeFromDividends(gogeDao);
     }
 
     function whitelistPinkSale(address _presaleAddress) external {
-        require(_msgSender() == DAO || _msgSender() == owner(), "Not authorized");
+        require(_msgSender() == gogeDao || _msgSender() == owner(), "GogeToken.sol::whitelistPinkSale() msg.sender is not owner or gogeDao");
+
         presaleAddress = _presaleAddress;
-        cakeDividendTracker.excludeFromDividends(address(_presaleAddress));
 
+        cakeDividendTracker.excludeFromDividends(address(_presaleAddress));
```

Comment:
Why is this being cast to address if it's already an address type?

Reply:
good catch. Changed that